cadquery/occ_impl/shapes.py

```diff
@@ -3984,3 +4094,882 @@ def edgesToWires(edges: Iterable[Edge], tol: float = 1e-6) -> List[Wire]:
+        yield s
+    elif t == "Edge":
+        yield Wire.assembleEdges((tcast(Edge, s),))
+    elif t == "Compound":
+        for el in s:
+            yield from _get_wires(el)
+    else:
+        raise ValueError(f"Required type(s): Edge, Wire; encountered {t}")
+
+
+def _get_edges(s: Shape) -> Iterable[Shape]:
+    """
+    Get wires or wires from edges
+    """
+
+    t = s.ShapeType()
+
+    if t == "Edge":
+        yield s
+    elif t == "Wire":
+        yield from _get_edges(s.edges())
+    elif t == "Compound":
+        for el in s:
+            yield from _get_edges(el)
+    else:
+        raise ValueError(f"Required type(s): Edge, Wire; encountered {t}")
+
+
+def _get_wire_lists(s: Sequence[Shape]) -> List[List[Wire]]:
+    """
+    Get lists or wires for sweeping or lofting
+    """
+
+    wire_lists: List[List[Wire]] = []
+
+    for el in s:
+        if not wire_lists:
+            wire_lists = [[w] for w in _get_wires(el)]
+        else:
+            for wire_list, w in zip(wire_lists, _get_wires(el)):
+                wire_list.append(w)
+
+    return wire_lists
+
+
+def _normalize(s: Shape) -> Shape:
+    """
+    Apply some normalizations:
+    - Shell with only one Face -> Face
+    """
+
+    t = s.ShapeType()
+    rv = s
+
+    if t == "Shell":
+        faces = s.Faces()
+        if len(faces) == 1:
+            rv = faces[0]
+    elif t == "Compound":
+        objs = list(s)
+        if len(objs) == 1:
+            rv = objs[0]
+
+    return rv
+
+
+def _compound_or_shape(s: Union[TopoDS_Shape, List[TopoDS_Shape]]) -> Shape:
+    """
+    Convert a list of TopoDS_Shape to a Shape or a Compound.
+    """
+
+    if isinstance(s, TopoDS_Shape):
+        rv = _normalize(Shape.cast(s))
+    elif len(s) == 1:
+        rv = _normalize(Shape.cast(s[0]))
+    else:
+        rv = Compound.makeCompound([_normalize(Shape.cast(el)) for el in s])
+
+    return rv
+
+
+def _pts_to_harray(ps: Sequence[VectorLike]) -> TColgp_HArray1OfPnt:
+    """
+    Convert a sequence of Vecotor to a TColgp harray (OCCT specific).
+    """
+
+    rv = TColgp_HArray1OfPnt(1, len(ps))
+
+    for i, p in enumerate(ps):
+        rv.SetValue(i + 1, Vector(p).toPnt())
+
+    return rv
+
+
+def _shapes_to_toptools_list(s: Iterable[Shape]) -> TopTools_ListOfShape:
+    """
+    Convert an itrable of Shape to a TopTools list (OCCT specific).
+    """
+
+    rv = TopTools_ListOfShape()
+
+    for el in s:
+        rv.Append(el.wrapped)
+
+    return rv
+
+
+#%% alternative constructors
+
+
+@multimethod
+def wire(*s: Shape) -> Shape:
+    """
+    Build wire from edges.
+    """
+
+    builder = BRepBuilderAPI_MakeWire()
+
+    edges = _shapes_to_toptools_list(e for el in s for e in _get_edges(el))
+    builder.Add(edges)
+
+    return _compound_or_shape(builder.Shape())
+
+
+@wire.register
+def wire(s: Sequence[Shape]) -> Shape:
+
+    return wire(*s)
+
+
+@multimethod
+def face(*s: Shape) -> Shape:
+    """
+    Builld face from edges or wires
```

Comment:
typo: Builld ->Build

Add period to face docstring?  Add period also to next functions shell, solid?